packages/sp/rest.ts

```diff
@@ -33,7 +33,7 @@ export class SPRest {
      * 
      * @param config The SharePoint configuration to apply
      */
-    public setup(config: SPConfiguration | ISPFXContext) {
+    public setup(config: SPConfiguration | ISPFXContext | any) {
```

Comment:
I am not sure about typing this param as any? Why not mark the property as optional in ISPFXContext? Putting any here makes it seem like you can just pass us anything and we will somehow figure it out when there are really only two cases covered. Thoughts?

Reply:
I've made some changes, and I agree. Let me know if the latest changes are good. I tested these changes on a SP2019 SPFx project. I also think we can remove both aadInfo & legacyPageContext from the pageContext object. I don't see any references to those in the pnpjs library?

Reply:
Yes, we can probably drop them. We used to use them but I think we stopped after we introduced the latest graph token stuff a while ago.